portal/src/ScreenNav.tsx

```diff
@@ -2,6 +2,27 @@ import React, { useMemo, useCallback, useContext } from "react";
 import { useNavigate, useLocation } from "react-router-dom";
 import { Context } from "@oursky/react-messageformat";
 import { Nav, INavLink, INavLinkGroup, INavProps } from "@fluentui/react";
+import { Location } from "history";
+
+function getAppRouterPath(location: Location) {
+  // app router -> /app/:appID/*
+  // discard first 3 segment (include leading slash)
+  return location.pathname.split("/").slice(3).join("/");
+}
+
+function getPath(url: string) {
+  // remove fragment
+  const path = new URL("scheme:" + url).pathname;
+  // remove leading trailing slash
+  const regexpMatch = /^\/?(.*[^/])\/?$/.exec(path);
```

Comment:
`/^\/?(.*[^/])\/?$/.exec("/")` is null.

I would write

```typescript
// "/" would become empty string
// Is this result we are expecting?
const pathWithoutLeadingTrailingSlash = path.replace(/^\//, "").replace(/\/$/, "")
```

Reply:
> "/" would become empty string
Is this result we are expecting?

I think we won't have "/" in navigation bar item, and removing slash is for consistency in comparison, so I think "/" becomes empty string is OK (it will work even if the path is "/"). 

> const pathWithoutLeadingTrailingSlash = path.replace(/^\//, "").replace(/\/$/, "")

This looks much nicer